### PR TITLE
[PW-1900] pay by link option

### PR DESCRIPTION
**Pull request by AlexandrosMor**
**Description**
Add a new payment method PayByLink into the Magento1 extension. This will be the only payment method avaialble after the june 2020 deadline because of Magento End Of Life.

**Tested scenarios**

**Fixed issue**:

### CODE REVIEW

File: app/code/community/Adyen/Payment/Model/Api.php [hunk 1]
```diff
@@ -899,4 +897,62 @@ public function setApplicationInfo($request)
...
+            $requestUrl = self::ENDPOINT_CHECKOUT_TEST . "/v41/paymentLinks";
+        } else {
+            $requestUrl = self::ENDPOINT_PROTOCOL .
+                $this->_helper()->getConfigData("live_endpoint_url_prefix") .
+                self::CHECKOUT_ENDPOINT_LIVE_SUFFIX . "/v41/paymentLinks";
+        }
+
+        $billingCountryCode = (is_object($order->getBillingAddress()) && $order->getBillingAddress()->getCountry() != "") ?
+            $order->getBillingAddress()->getCountry() :
+            false;
```
Comment (rikterbeek) [inline app/code/community/Adyen/Payment/Model/Api.php]: add some defence programming if billingAddress does not exists.

File: app/design/frontend/base/default/template/adyen/form/pay_by_link.phtml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+<?php
+/**
+ * @var $this Adyen_Payment_Block_Form_PayByLink
+ */
+?>
```
Comment (rikterbeek) [inline app/design/frontend/base/default/template/adyen/form/pay_by_link.phtml]: Add a comment that you will be redirected to an external page. Make this a translatable text.

File: app/design/frontend/base/default/template/adyen/info/pay_by_link.phtml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+ * @category   Payment Gateway
+ * @package    Adyen_Payment
+ * @author     Adyen
+ * @property   Adyen B.V
+ * @copyright  Copyright (c) 2014 Adyen BV (http://www.adyen.com)
+ */
+?>
+<?php echo $this->htmlEscape($this->getMethod()->getTitle()) ?><br/>
+    <a target="_blank"
+       href="<?php echo $this->getMethod()->getInfoInstance()->getAdditionalInformation('payment_url'); ?>"><?php echo $this->__("Click here to pay for your purchase"); ?></a>
```
Comment (rikterbeek) [inline app/design/frontend/base/default/template/adyen/info/pay_by_link.phtml]: this should go to the form. Info should contain same info as the paybymail payment method.

**Comment by AlexandrosMor** [comment]
@msilvagarcia 
Thank you for the feedback, It is not affecting other parts as the method is used as follows:
$this->_initOrder();
$order = $this->_order;

**Comment by seansan** [comment]
WHat do you mean by " This will be the **only payment** method avaialble after the june 2020 deadline" ...

**Comment by Aleffio** [comment]
Hi @seansan, 
thanks for your feedback,

we adjusted the description of the pull request.
We will post more information before the end of the month on our documentation pages,

Kind Regards,
Alessio
Adyen

**Pull request merged**